src/main/java/de/tum/in/www1/artemis/repository/TeamRepository.java

```diff
@@ -122,4 +132,8 @@ private List<Pair<User, Team>> findStudentTeamConflicts(Exercise exercise, Team
     default Team findByIdElseThrow(long teamId) throws EntityNotFoundException {
         return findById(teamId).orElseThrow(() -> new EntityNotFoundException("Team", teamId));
     }
+
+    default Team findWithStudentsByIdElseThrow(long teamId) throws EntityNotFoundException {
+        return findWithStudentsById(teamId).orElseThrow(() -> new EntityNotFoundException("Team", teamId));
+    }
```

Comment:
probably move this to it's relative `findWithStudentsById`

Reply:
It's not uncommon in our system to have the default implementations separate from the interface methods.